modules/tooling-support/src/main/java/org/mule/runtime/module/tooling/internal/artifact/metadata/MetadataKeyDeclarationResolver.java

```diff
@@ -150,7 +152,12 @@ public MetadataKeyResult resolveKeyResult() {
             .orElseThrow(() -> new MuleRuntimeException(createStaticMessage("Could not find parameter with name: %s in parameter group: %s",
                                                                             parameterName, parameterGroupName)));
         if (parameterModel.getModelProperty(MetadataKeyPartModelProperty.class).isPresent()) {
-          parametersMap.put(parameterName, extractSimpleValue(parameterElement.getValue()));
+          String value = extractSimpleValue(parameterElement.getValue());
+          if (isExpression(value)) {
```

Comment:
Same here. I just realized that I should first check if the parameterModel.isSupportExpression() == true before checking the String content.